linode/token/framework_resource.go

```diff
@@ -180,48 +180,32 @@ func (r *Resource) Update(
 	req resource.UpdateRequest,
 	resp *resource.UpdateResponse,
 ) {
-	var data ResourceModel
-	var tokenIDString string
-	resp.Diagnostics.Append(
-		req.State.GetAttribute(ctx, path.Root("id"), &tokenIDString)...,
-	)
-	if resp.Diagnostics.HasError() {
-		return
-	}
+	var state ResourceModel
+	var plan ResourceModel
 
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 
-	tokenID := int(helper.StringToInt64(tokenIDString, resp.Diagnostics))
-	if resp.Diagnostics.HasError() {
-		return
-	}
+	resourceUpdated := !state.Label.Equal(plan.Label)
 
-	client := r.client
-	token, err := client.GetToken(ctx, tokenID)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			fmt.Sprintf("Failed to get the token with id %v", tokenID),
-			err.Error(),
-		)
-		return
-	}
-
-	updateOpts := token.GetUpdateOptions()
-	plannedTokenLabel := data.Label.ValueString()
+	if resourceUpdated {
```

Comment:
I think you can directly compare these two values instead of storing it as a flag:
```suggestion
	if !state.Label.Equal(plan.Label) {
	    ...
	}
```

Reply:
Thank you for the suggestion!